app/models/katello/candlepin/content.rb

```diff
@@ -52,28 +52,40 @@ def find_repository
-                                 :product => product,
-                                 :pulp_id => pulp_id,
-                                 :cp_label => content.label,
-                                 :content_id => content.id,
-                                 :arch => arch,
-                                 :major => major,
-                                 :minor => minor,
-                                 :relative_path => relative_path,
-                                 :name => name,
-                                 :label => label,
-                                 :url => feed_url,
-                                 :feed_ca => ca,
-                                 :feed_cert => product.certificate,
-                                 :feed_key => product.key,
-                                 :content_type => katello_content_type,
-                                 :preserve_metadata => true, #preserve repo metadata when importing from cp
-                                 :unprotected => unprotected?,
-                                 :content_view_version => product.organization.library.default_content_view_version)
+        repository = Repository.new(
+                                  :environment => product.organization.library,
+                                  :product => product,
+                                  :pulp_id => pulp_id,
+                                  :cp_label => content.label,
+                                  :content_id => content.id,
+                                  :arch => arch,
+                                  :major => major,
+                                  :minor => minor,
+                                  :relative_path => relative_path,
+                                  :name => name,
+                                  :label => label,
+                                  :url => feed_url,
+                                  :feed_ca => ca,
+                                  :feed_cert => product.certificate,
+                                  :feed_key => product.key,
+                                  :content_type => katello_content_type,
+                                  :preserve_metadata => true, #preserve repo metadata when importing from cp
+                                  :unprotected => unprotected?,
+                                  :content_view_version => product.organization.
+                                                          library.default_content_view_version
+        )
+
+        if katello_content_type == Repository::OSTREE_TYPE
+          # TODO
+          # add a hard coded branch for now
+          # we will add pull all once the summary files functionality is available
+          repository.ostree_branches.new(:name => "rhel-atomic-host/7/x86_64/standard", :repository => repository)
+        end
+        repository
       end
 
       def validate!
+        return if katello_content_type == Repository::OSTREE_TYPE
```

Comment:
point of a method ending with ! is that its return value is void..

look at the implementation of validate! in that file in general. I think it will make that clear .. https://github.com/parthaa/katello/blob/cdn/app/models/katello/candlepin/content.rb#L88